UM/Math/Polygon.py

```diff
@@ -182,6 +182,11 @@ def intersectsPolygon(self, other: "Polygon") -> Optional[Tuple[float, float]]:
             return None
 
         polygon_intersection = polygon_me.intersection(polygon_other)
+        if polygon_intersection.is_empty:
+            return None
+        if not polygon_intersection.is_valid:
+            return None
+
```

Comment:
I think we should move these checks into the `if polygon_intersection` check down below. We could even make it part of the `if` condition itself, like so:

    if polygon_intersection and not polygon_intersection.is_empty and polygon_intersection.is_valid: